libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1375,6 +1375,20 @@ bool ExpressionCompiler::visit(MemberAccess const& _memberAccess)
 			setLValue<MemoryItem>(_memberAccess, *_memberAccess.annotation().type);
 			break;
 		}
+		case DataLocation::CallData:
+		{
+			solUnimplementedAssert(!type.isDynamicallyEncoded(), "");
+			m_context << type.calldataOffsetOfMember(member) << Instruction::ADD;
+			// For non-value types the calldata offset is returned directly.
+			if (_memberAccess.annotation().type->isValueType())
+			{
+				m_context << Instruction::DUP1;
+				m_context << type.calldataEncodedSize(true);
+				m_context << Instruction::ADD;
+				CompilerUtils(m_context).abiDecode({_memberAccess.annotation().type}, false);
```

Comment:
I'm wondering whether this way of constructing temporary ``TypePointers`` and passing it to ``abiDecode`` and further to ``ABIFunctions::tupleDecoder`` will be subject to the problem described in #5948... thinking about it, I'm a bit surprised that it **doesn't** break as it is...

Reply:
Obsolete - won't be a problem.